### PR TITLE
Added a ThreadPool to limit merging of large results.

### PR DESCRIPTION
**Pull request by jgates108**

### CODE REVIEW

File: core/modules/rproc/InfileMerger.cc [hunk 1]
```diff
@@ -171,10 +148,53 @@ bool InfileMerger::merge(std::shared_ptr<proto::WorkerResponse> response) {
...
     // Nothing to do if size is zero.
-    if (response->result.row_size() != 0) {
-        return _merge(response);
+    if (response->result.row_size() == 0) {
+        return true;
     }
-    return true;
+
+    std::string virtFile = _infileMgr.prepareSrc(newProtoRowBuffer(response->result));
+    std::string infileStatement = sql::formLoadInfile(_mergeTable, virtFile);
```
Comment (andy-slac) [inline core/modules/rproc/InfileMerger.cc]: make this and above `const`

File: core/modules/rproc/InfileMerger.h [hunk 2]
```diff
@@ -157,9 +161,9 @@ class InfileMerger {
     std::mutex _mysqlMutex;
     lsst::qserv::mysql::LocalInfile::Mgr _infileMgr;
 
-    // The limited size pool will keep this query from using up all the czar's time
-    // when only a couple of threads can effective write to a single table.
-    util::ThreadPool::Ptr _largeResultPool = util::ThreadPool::newThreadPool(3, nullptr);
+    // The limited size pool will keep large queries from using up all the czar's time.
+    static util::ThreadPool::Ptr _largeResultPool;
+    static std::mutex largeResultPoolMutex;
```
Comment (andy-slac) [inline core/modules/rproc/InfileMerger.h]: Underscore prefix still needed for this name? I somewhat prefer not to pollute class declaration with static members but hide them in anonymous namespace.

File: core/modules/rproc/InfileMerger.cc [hunk 3]
```diff
@@ -109,13 +111,29 @@ InfileMerger::InfileMerger(InfileMergerConfig const& c)
...
 }
 
 InfileMerger::~InfileMerger() {
-    _largeResultPool->endAll();
+}
+
+
+int InfileMerger::setLargeResultPoolSize(int size) {
+    std::lock_guard<std::mutex> lock(largeResultPoolMutex);
+    size = std::min(1, size); // size must be at least 1
```
Comment (andy-slac) [inline core/modules/rproc/InfileMerger.cc]: `std::max`?

Comment (jgates108) [inline core/modules/rproc/InfileMerger.cc]: oops

File: core/modules/rproc/InfileMerger.cc [hunk 4]
```diff
@@ -109,13 +111,29 @@ InfileMerger::InfileMerger(InfileMergerConfig const& c)
...
+    if (_largeResultPool == nullptr) {
+        _largeResultPool = util::ThreadPool::newThreadPool(size, nullptr);
+    }
+    _largeResultPool->resize(size);
```
Comment (andy-slac) [inline core/modules/rproc/InfileMerger.cc]: This is not strictly needed when you have just created thread pool?

File: core/modules/czar/Czar.h [hunk 5]
```diff
@@ -104,6 +104,7 @@ class Czar {
     std::unique_ptr<ccontrol::UserQueryFactory> _uqFactory;
     ClientToQuery _clientToQuery;       ///< maps client ID to query
     std::mutex _mutex;                  ///< protects both _uqFactory and _clientToQuery
+
```
Comment (andy-slac) [inline core/modules/czar/Czar.h]: I guess this is leftover from you experiments with adding data members here, may be better not to touch this file at all in this commit?

**Pull request merged**